pkg/adapter/mtping/runner.go

```diff
@@ -192,9 +192,11 @@ func (a *cronJobsRunner) newPingSourceClient(source *sourcesv1.PingSource) (adap
 	var env adapter.EnvConfig
 	if a.clientConfig.Env != nil {
 		env = adapter.EnvConfig{
-			Namespace:      source.GetNamespace(),
-			Name:           a.clientConfig.Env.GetName(),
-			EnvSinkTimeout: fmt.Sprintf("%d", a.clientConfig.Env.GetSinktimeout()),
+			Namespace:              source.GetNamespace(),
+			Name:                   a.clientConfig.Env.GetName(),
+			EnvSinkTimeout:         fmt.Sprintf("%d", a.clientConfig.Env.GetSinktimeout()),
+			Audience:               source.Status.SinkAudience,
+			OIDCServiceAccountName: source.Status.Auth.ServiceAccountName,
```

Comment:
When OIDC is not enabled, `source.Status.Auth` is nil. So you should check it before setting, to avoid running into nil panics.